news/create-newsfile.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+#############################################################################
+# Copyright (c) 2020 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+import re
+import sys
+from argparse import ArgumentParser
+from datetime import datetime, timedelta, timezone
+from pathlib import Path
+from subprocess import PIPE, Popen
+
+news_dir = Path(__file__).resolve().parent
+root_dir = news_dir.parent
+newsfile = root_dir / 'NEWS.md'
+
+
+def print_usage_if_needed():
+    ArgumentParser(usage="\rCreates NEWS.md file from the entries in the news/ folder.\n"
+                         "It also deletes the entry files.").parse_args()
+
+
+def create_block(block_name, files):
+    block = '## {}\n\n'.format(block_name)
+    for f in files:
+        entry = ''
+        try:
+            pr_id = re.findall(r'\d+.md$', f.name)[0][:-3]
+        except IndexError:
+            sys.exit('Invalid filename: {}'.format(f.name))
+        entry += ' * {} (#{})'.format(f.read_text().rstrip(), pr_id)
+        entry = entry.replace('\n', '\n   ')
+        block += entry + '\n'
+    block += '\n'
+    return block
+
+
+def get_version_file_change_date():
+    proc = Popen(r'git log --format=%cd -n 1 VERSION'.split(), cwd=str(root_dir), stdout=PIPE)
+    stdout, _ = proc.communicate()
+    return datetime.strptime(stdout.decode().rstrip(), r'%a %b %d %H:%M:%S %Y %z')
+
+
+def older_than_2_weeks(change_date):
+    current_date = datetime.now(timezone.utc)
+    return current_date - timedelta(weeks=2) > change_date
+
+
+def create_version():
+    change_date = get_version_file_change_date()
+    if (older_than_2_weeks(change_date)):
```

Comment:
Could you explain the use case for this 2 weeks check ? (I would keep it simple and use the version file.)

Reply:
Sure.
The script should be run, after bumping the version in the `VERSION` file. If this heuristic observes, that the `VERSION` file was not updated in a git commit in the last 2 weeks, it probably contains the previously released version, which is not good for our new newsfile.

Reply:
I would not mix up the release process and a newsfile generator script, as that just complicates this script. (Not a change request at all.) 
But if you still persist with it, you might additionally consider check agains the `NEWS.md` file version instead (you can get the one that is stored in git so working dir modification does not take effect in case of re-executing the script multiple time).

Reply:
Comparing to `git show HEAD:NEWS.md` should be enough right?

Reply:
That's the one I was referring to.

Reply:
Thanks, fixed. This is much better now :)